src/err/mod.rs

```diff
@@ -439,9 +439,8 @@ impl PyErr {
     where
         T: ToPyObject,
     {
-        unsafe {
-            ffi::PyErr_GivenExceptionMatches(self.type_ptr(py), exc.to_object(py).as_ptr()) != 0
-        }
+        let exc = exc.to_object(py);
+        unsafe { ffi::PyErr_GivenExceptionMatches(self.type_ptr(py), exc.as_ptr()) != 0 }
```

Comment:
Candidate for an non-generic inner impl or too short to matter?

Reply:
Done 👍